### PR TITLE
training/abstractInterfaceConcept

### PR DESCRIPTION
**Pull request by shrikanttechincal**
15.Create one class having 1 method with different parameters like fullName(String firstName), fullName(String firstName, String 
    lastName),  fullName(String firstName, String lastName, String middleName), call all this method method and print in the 
   console.
   Note - Method overloading concept

16. Create an abstract class WillDoEverthing in that create one method sum(int a), Call that sum() method from Class Yes1 and 
     do addition of three numbers.
    Note - Use abstraction concept

17.   Create an abstract class WillDoEverthing in that create one method sum(int a), Call that sum() method from Class Yes1 and 
       do addition of three numbers.
      Note - Use Interface concept

### CODE REVIEW

File: src/test/java/com/training/assignment/tests/AssignmentTest.java [hunk 1]
```diff
@@ -139,4 +142,26 @@ public void testOopsConcept() {
         softAssert.assertThat(director.getWorkDetail())
                 .as("Work of actor and director differs").isEqualTo("Acting Directing");
     }
+
+    @Test
+    public void testAbstractInterfaceConcept() {
+        Employee employee = new Employee();
+        Yes1 yes1 = new Yes1();
```
Comment (PranavHanwante) [inline src/test/java/com/training/assignment/tests/AssignmentTest.java]: Line space

File: src/test/java/com/training/assignment/tests/AssignmentTest.java [hunk 2]
```diff
@@ -139,4 +142,26 @@ public void testOopsConcept() {
...
+        Yes2 yes2 = new Yes2();
+
+        softAssert.assertThat(employee.getName("Shrikant"))
+                .as("Employee name doesn't match,try something else").isEqualTo("Shrikant");
+
+        softAssert.assertThat(employee.getName("Shrikant", "Sajjanshetti"))
+                .as("Employee name doesnt match,try something else").isEqualTo("Shrikant Sajjanshetti");
+
+        softAssert.assertThat(employee.getName("Shrikant", "Sajjanshetti", "Umakant"))
+                .as("Employee name doesnt match,try something else").isEqualTo("Shrikant Sajjanshetti Umakant");
```
Comment (PranavHanwante) [inline src/test/java/com/training/assignment/tests/AssignmentTest.java]: So when you are hard coding same value more than one time, then you should define it in a variable, in your case Shrikant is hardcoded more than one time, so define it in a varible.
Check remaining too.

File: src/test/java/com/training/assignment/tests/AssignmentTest.java [hunk 3]
```diff
@@ -139,4 +142,32 @@ public void testOopsConcept() {
...
+    public void testAbstractInterfaceConcept() {
+        Employee employee = new Employee();
+
+        Yes1 yes1 = new Yes1();
+
+        Yes2 yes2 = new Yes2();
+
+        String firstName="Shrikant";
+        String lastName="Sajjanshetti";
+        String middleName="Umakant";
```
Comment (PranavHanwante) [inline src/test/java/com/training/assignment/tests/AssignmentTest.java]: hard code this, Umakant, as we are using this one time inside the method.

File: src/test/java/com/training/assignment/tests/AssignmentTest.java [hunk 4]
```diff
@@ -139,4 +142,32 @@ public void testOopsConcept() {
...
+        softAssert.assertThat(employee.getName(firstName, lastName))
+                .as("Employee name doesnt match,try something else").isEqualTo("Shrikant Sajjanshetti");
+
+        softAssert.assertThat(employee.getName(firstName, lastName, middleName))
+                .as("Employee name doesnt match,try something else").isEqualTo("Shrikant Sajjanshetti Umakant");
+
+        softAssert.assertThat(yes1.getSum(3))
+                .as("Sum of three number is incorrect").isEqualTo(33);
+
+        softAssert.assertThat(yes2.getSum(5))
```
Comment (PranavHanwante) [inline src/test/java/com/training/assignment/tests/AssignmentTest.java]: Direclty use, softAssert.assertThat(new Yes1().getSum(3))
                .as("Sum of three number is incorrect").isEqualTo(33);

And for below code too

**Review by PranavHanwante — APPROVED** [review]

**Review by akhil7000 — APPROVED** [review]

**Pull request merged**